pkg/addr/host.go

```diff
@@ -88,22 +89,28 @@ func MustParseHost(s string) Host {
 }
 
 // HostIP returns a Host address representing ip, with type HostTypeIP.
+// @ decreases
 func HostIP(ip netip.Addr) Host {
 	return Host{t: HostTypeIP, ip: ip}
 }
 
 // HostSvc returns a Host address representing svc, with type HostTypeSVC.
+// @ decreases
 func HostSVC(svc SVC) Host {
 	return Host{t: HostTypeSVC, svc: svc}
 }
 
 // Type returns the type of the address represented by h.
+// @ pure
+// @ decreases
 func (h Host) Type() HostAddrType {
 	return h.t
 }
 
 // IP returns the IP address represented by h.
 // Panics if h.Type() is not HostTypeIP.
+// @ requires h.t == HostTypeIP
```

Comment:
Notice that this is a spec for a public function (thus, it needs to be understandable by clients), but it mentions a private field (which clients are not aware of). Thus, this precondition cannot be really be ever established by a client - we should change the precondition to only use public members, like the `Type()` function instead.

Reply:
Done.